File: app/src/main/java/com/example/sqliteapp/MainActivity.java
package com.example.sqliteapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    //references to layout controls
    Button btn_login, btn_toDB;
    EditText et_fName, et_sName, et_IDNP;
    DatabaseHelper databaseHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btn_login = findViewById(R.id.btn_login);
        btn_toDB = findViewById(R.id.btn_toDB);
        et_fName = findViewById(R.id.et_firstName);
        et_sName = findViewById(R.id.et_secondName);
        et_IDNP = findViewById(R.id.et_IDNP);
        databaseHelper = new DatabaseHelper(MainActivity.this);

        //button listeners
        btn_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (et_fName.getText().toString().length() == 0 || et_sName.getText().toString().length() == 0  || et_IDNP.getText().toString().length() == 0 ){
                    Toast.makeText(MainActivity.this, "Please fill all the fields.", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (databaseHelper.checkLogin(new PersonModel(et_fName.getText().toString(), et_sName.getText().toString(), et_IDNP.getText().toString()))){
                    Intent intent = new Intent(MainActivity.this, Logged_in_2nd.class);
                    String[] toPut = databaseHelper.getFromDb(et_IDNP.getText().toString());
                    intent.putExtra("VaccinationType", toPut[0]);
                    intent.putExtra("VaccinationDate", toPut[1]);
                    intent.putExtra("IDNP", toPut[2]);
                    startActivity(intent);
                }
                else {
                    Toast.makeText(MainActivity.this, "No account found", Toast.LENGTH_SHORT).show();
                }

            }
        });

        btn_toDB.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(MainActivity.this, DatabaseActivity.class);
                startActivity(intent);


            }
        });


    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
        databaseHelper = null;
    }


}

File: app/src/main/java/com/example/sqliteapp/activities/main/MainActivity.kt
package com.example.sqliteapp.activities.main

import android.content.Intent
import android.os.Bundle
import android.os.Looper
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.example.sqliteapp.activities.vaccdb.DatabaseActivity
import com.example.sqliteapp.activities.testdb.DatabaseTestActivity
import com.example.sqliteapp.activities.login.LoggedInActivity
import com.example.sqliteapp.databinding.ActivityMainBinding
import kotlinx.coroutines.*
import java.lang.IllegalArgumentException


//the first activity you see upon starting the app,
open class MainActivity : AppCompatActivity() {
    //use binding for more efficient layout control
    private lateinit var mBinding: ActivityMainBinding
    private val binding get() = mBinding
    private val handler = CoroutineExceptionHandler{_, exception ->
        if (Looper.myLooper() == null) {
            Looper.prepare()
            Looper.loop()
        }
        Log.i("Coroutine:", "Error", exception)
        Toast.makeText(this, "Error: ${exception.cause}", Toast.LENGTH_SHORT).show()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = ActivityMainBinding.inflate(this.layoutInflater)
        val view = mBinding.root
        setContentView(view)
        val mainActivityViewModel = ViewModelProvider(this).get(MainActivityViewModel::class.java)

        binding.btnLogin.setOnClickListener {
            try {
                mainActivityViewModel.onClickedLogin(
                        binding.etFirstName.text.toString(),
                        binding.etSecondName.text.toString(),
                        binding.etIDNP.text.toString(),
                        handler
                )
                val intent = Intent(this@MainActivity, LoggedInActivity::class.java)
                intent.putExtra("fn", binding.etFirstName.text.toString())
                intent.putExtra("IDNP", binding.etIDNP.text.toString())
                intent.putExtra("sn", binding.etSecondName.text.toString())
                startActivity(intent)
            } catch (exception: IllegalArgumentException){
                Toast.makeText(this@MainActivity, "Please fill all the fields.", Toast.LENGTH_SHORT).show()
            } catch (exception: NoSuchElementException){
                Toast.makeText(this@MainActivity, "No account found", Toast.LENGTH_SHORT).show()
            }
        }
        //go to database population screen
        binding.btnToDB.setOnClickListener {
            val intent = Intent(this@MainActivity, DatabaseActivity::class.java)
            startActivity(intent)
        }
        binding.btnToTestDb.setOnClickListener{
            val intent = Intent(this@MainActivity, DatabaseTestActivity::class.java)
            startActivity(intent)
        }
    }
}